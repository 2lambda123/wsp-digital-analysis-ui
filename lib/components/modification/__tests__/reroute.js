// @flow
import React from 'react'

import {
  MAP_STATE_SELECT_FROM_STOP,
  MAP_STATE_SELECT_TO_STOP,
  REROUTE
} from '../../../constants'
import Reroute from '../reroute'
import {
  mockFeed,
  mockModification,
  mockWithProvider,
  mockRoute
} from '../../../utils/mock-data'

const mockRerouteModification = {
  ...mockModification,
  dwellTime: 0,
  fromStop: '1',
  routes: [mockRoute.route_id],
  segmentSpeeds: [],
  segments: [],
  toStop: '2',
  type: REROUTE
}

describe('Component > Modification > Reroute', () => {
  it('renders correctly', () => {
    const props = {
      feeds: [mockFeed],
      mapState: { state: 'reroute' },
      modification: mockRerouteModification,
      qualifiedStops: [],
      routePatterns: [],
      selectedFeed: mockFeed,
      segmentDistances: [],
      setMapState: jest.fn(),
      stops: [],
      update: jest.fn(),
      updateAndRetrieveFeedData: jest.fn()
    }
    const mock = mockWithProvider(<Reroute {...props} />)
    expect(mock.snapshot()).toMatchSnapshot()

    mock.wrapper.find('a[title="Select from stop"]').simulate('click')
    expect(props.setMapState.mock.calls[0][0]).toEqual({
      allowExtend: false,
      extendFromEnd: true,
      state: MAP_STATE_SELECT_FROM_STOP
    })

    mock.wrapper.find('a[title="Select to stop"]').simulate('click')
    expect(props.setMapState.mock.calls[1][0]).toEqual({
      allowExtend: false,
      extendFromEnd: false,
      state: MAP_STATE_SELECT_TO_STOP
    })

    mock.wrapper.find('a[title="Clear from stop"]').simulate('click')
    expect(props.update.mock.calls[0][0]).toEqual({
      fromStop: null,
      toStop: mockRerouteModification.toStop,
      segments: []
    })

    mock.wrapper.find('a[title="Clear to stop"]').simulate('click')
    expect(props.update.mock.calls[1][0]).toEqual({
      fromStop: mockRerouteModification.fromStop,
      toStop: null,
      segments: []
    })
  })
})
