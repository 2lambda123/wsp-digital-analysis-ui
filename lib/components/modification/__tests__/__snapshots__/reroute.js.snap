// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component > Modification > Reroute renders correctly 1`] = `
<div>
  <div>
    <div>
      <div
        className="form-group"
      >
        <div
          className="Select Select--single is-searchable"
          style={undefined}
        >
          <div
            className="Select-control"
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            style={undefined}
          >
            <span
              className="Select-multi-value-wrapper"
              id="react-select-2--value"
            >
              <div
                className="Select-placeholder"
              >
                Select feed
              </div>
            </span>
            <span
              className="Select-arrow-zone"
              onMouseDown={[Function]}
            >
              <span
                className="Select-arrow"
                onMouseDown={[Function]}
              />
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="form-group"
    >
      <div
        className="Select Select--multi is-searchable"
        style={undefined}
      >
        <div
          className="Select-control"
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={undefined}
        >
          <span
            className="Select-multi-value-wrapper"
            id="react-select-3--value"
          >
            <div
              className="Select-placeholder"
            >
              Select patterns
            </div>
          </span>
          <span
            aria-label="Clear all"
            className="Select-clear-zone"
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            title="Clear all"
          >
            <span
              className="Select-clear"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "&times;",
                }
              }
            />
          </span>
          <span
            className="Select-arrow-zone"
            onMouseDown={[Function]}
          >
            <span
              className="Select-arrow"
              onMouseDown={[Function]}
            />
          </span>
        </div>
      </div>
    </div>
  </div>
  <div>
    <a
      className="btn btn-warning btn-block"
      href="#"
      onClick={[Function]}
      tabIndex={0}
      target={undefined}
    >
      <i
        className="fa fa-stop-circle fa-fw "
      />
       
      Stop editing
    </a>
    <div
      className="checkbox"
    >
      <label
        htmlFor="create-stops-automatically-input-0"
      >
        <input
          defaultChecked={true}
          id="create-stops-automatically-input-0"
          onChange={[Function]}
          type="checkbox"
        />
         
        Create stops automatically
      </label>
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="stop-spacing-input-1"
      >
        Stop spacing
      </label>
      <div
        className="InputWithUnits"
      >
        <input
          className="form-control"
          id="stop-spacing-input-1"
          min={0}
          name={undefined}
          onChange={[Function]}
          onWheel={[Function]}
          placeholder="meters"
          type="number"
          value={400}
        />
        <span
          className="InputUnits"
        >
          meters
        </span>
      </div>
    </div>
    <div
      className="checkbox"
    >
      <label
        htmlFor="bidirectional-input-2"
      >
        <input
          checked={undefined}
          id="bidirectional-input-2"
          onChange={[Function]}
          type="checkbox"
        />
         
        Bidirectional
      </label>
    </div>
    <div
      className="checkbox"
    >
      <label
        htmlFor="extend-from-end-input-3"
      >
        <input
          defaultChecked={true}
          id="extend-from-end-input-3"
          onChange={[Function]}
          type="checkbox"
        />
         
        Extend from end
      </label>
    </div>
    <div
      className="checkbox"
    >
      <label
        htmlFor="follow-streets-input-4"
      >
        <input
          defaultChecked={undefined}
          id="follow-streets-input-4"
          onChange={[Function]}
          type="checkbox"
        />
         
        Follow streets
      </label>
    </div>
  </div>
  <div>
    <div
      className="form-group"
    >
      <label
        htmlFor="dwell-time-input-5"
      >
        Dwell time
      </label>
      <div
        className="InputWithUnits"
      >
        <input
          className="form-control"
          id="dwell-time-input-5"
          min={0}
          name={undefined}
          onChange={[Function]}
          onWheel={[Function]}
          placeholder="seconds"
          type="number"
          value={0}
        />
        <span
          className="InputUnits"
        >
          seconds
        </span>
      </div>
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="average-segment-speed-input-6"
      >
        Average segment speed
      </label>
      <div
        className="InputWithUnits"
      >
        <input
          className="form-control"
          id="average-segment-speed-input-6"
          min={1}
          name={undefined}
          onChange={[Function]}
          onWheel={[Function]}
          placeholder="km/h"
          type="number"
          value={15}
        />
        <span
          className="InputUnits"
        >
          km/h
        </span>
      </div>
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="total-travel-time-input-7"
      >
        Total travel time
      </label>
      <div
        className="InputWithUnits"
      >
        <input
          className="form-control"
          id="total-travel-time-input-7"
          min={1}
          name={undefined}
          onChange={[Function]}
          onWheel={[Function]}
          placeholder="minutes"
          type="number"
          value={0}
        />
        <span
          className="InputUnits"
        >
          minutes
        </span>
      </div>
    </div>
    <div>
      <div
        aria-level={3}
        role="heading"
      >
        <a
          aria-expanded={false}
          className="CollapsibleButton"
          onClick={[Function]}
          role="button"
          tabIndex={0}
        >
          <i
            className="fa fa-caret-right fa-fw "
          />
          <strong>
            Set individual segment speeds
          </strong>
        </a>
      </div>
    </div>
  </div>
</div>
`;
